File: app/src/main/java/com/babestudios/companyinfouk/data/model/company/Company.java

package com.babestudios.companyinfouk.data.model.company;

import java.util.ArrayList;
import java.util.List;

import com.google.gson.annotations.SerializedName;

public class Company {

	@SerializedName("accounts")
	public Accounts accounts;

	@SerializedName("annual_return")
	public AnnualReturn annualReturn;

	@SerializedName("branch_company_details")
	public BranchCompanyDetails branchCompanyDetails;

	@SerializedName("can_file")
	public String canFile;

	@SerializedName("company_name")
	public String companyName;

	@SerializedName("company_number")
	public String companyNumber;

	@SerializedName("company_status")
	public String companyStatus;

	@SerializedName("company_status_detail")
	public String companyStatusDetail;

	@SerializedName("confirmation_statement")
	public ConfirmationStatement confirmationStatement;

	@SerializedName("date_of_cessation")
	public String dateOfCessation;

	@SerializedName("date_of_creation")
	public String dateOfCreation;

	@SerializedName("etag")
	public String etag;

	@SerializedName("foreign_company_details")
	public ForeignCompanyDetails foreignCompanyDetails;

	@SerializedName("has_been_liquidated")
	public String hasBeenLiquidated;

	@SerializedName("has_charges")
	public String hasCharges;

	@SerializedName("has_insolvency_history")
	public String hasInsolvencyHistory;

	@SerializedName("is_community_interest_company")
	public String isCommunityInterestCompany;

	@SerializedName("jurisdiction")
	public String jurisdiction;

	@SerializedName("last_full_members_list_date")
	public String lastFullMembersListDate;

	@SerializedName("links")
	public CompanyLinks links;

	@SerializedName("partial_data_available")
	public String partialDataAvailable;

	@SerializedName("previous_company_names")
	public List<PreviousCompanyName> previousCompanyNames = new ArrayList<PreviousCompanyName>();

	@SerializedName("registered_office_address")
	public RegisteredOfficeAddress registeredOfficeAddress;

	@SerializedName("registered_office_is_in_dispute")
	public String registeredOfficeIsInDispute;

	@SerializedName("sic_codes")
	public List<String> sicCodes = new ArrayList<String>();

	@SerializedName("type")
	public String type;

	@SerializedName("undeliverable_registered_office_address")
	public String undeliverableRegisteredOfficeAddress;

}

File: data/src/main/java/com/babestudios/companyinfouk/data/model/company/Company.kt
package com.babestudios.companyinfouk.data.model.company

import android.os.Parcelable
import com.babestudios.companyinfouk.data.model.common.Address
import java.util.ArrayList

import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
class Company(

		@SerializedName("accounts")
		var accounts: Accounts? = null,

		@SerializedName("annual_return")
		var annualReturn: AnnualReturn? = null,

		@SerializedName("branch_company_details")
		var branchCompanyDetails: BranchCompanyDetails? = null,

		@SerializedName("can_file")
		var canFile: String? = null,

		@SerializedName("company_name")
		var companyName: String? = null,

		@SerializedName("company_number")
		var companyNumber: String? = null,

		@SerializedName("company_status")
		var companyStatus: String? = null,

		@SerializedName("company_status_detail")
		var companyStatusDetail: String? = null,

		@SerializedName("confirmation_statement")
		var confirmationStatement: ConfirmationStatement? = null,

		@SerializedName("date_of_cessation")
		var dateOfCessation: String? = null,

		@SerializedName("date_of_creation")
		var dateOfCreation: String? = null,

		@SerializedName("etag")
		var etag: String? = null,

		@SerializedName("foreign_company_details")
		var foreignCompanyDetails: ForeignCompanyDetails? = null,

		@SerializedName("has_been_liquidated")
		var hasBeenLiquidated: String? = null,

		@SerializedName("has_charges")
		var hasCharges: Boolean = true,

		@SerializedName("has_insolvency_history")
		var hasInsolvencyHistory: Boolean = true,

		@SerializedName("is_community_interest_company")
		var isCommunityInterestCompany: String? = null,

		@SerializedName("jurisdiction")
		var jurisdiction: String? = null,

		@SerializedName("last_full_members_list_date")
		var lastFullMembersListDate: String? = null,

		@SerializedName("links")
		var links: CompanyLinks? = null,

		@SerializedName("partial_data_available")
		var partialDataAvailable: String? = null,

		@SerializedName("previous_company_names")
		var previousCompanyNames: List<PreviousCompanyName> = ArrayList(),

		@SerializedName("registered_office_address")
		var registeredOfficeAddress: Address? = null,

		@SerializedName("registered_office_is_in_dispute")
		var registeredOfficeIsInDispute: String? = null,

		@SerializedName("sic_codes")
		var sicCodes: List<String> = ArrayList(),

		@SerializedName("type")
		var type: String? = null,

		@SerializedName("undeliverable_registered_office_address")
		var undeliverableRegisteredOfficeAddress: String? = null
) : Parcelable
